### DISCUSSION TITLE
tslib outdated preventing the use of spread syntax?

### DISCUSSION THREAD

**Discussion by stephencweiss** [jaredpalmer/tsdx#1081] [Q&A]
I'm trying to use a spread operator on an array in my TSDX library and am getting the following error:

![Screen Shot 2021-09-21 at 2 02 24 PM](https://user-images.githubusercontent.com/39878535/134231940-32673fc2-10ac-4d21-9b5e-fea2213f0df2.png)

The error in full:
```shell
This syntax requires an imported helper named '__spreadArray' with 3 parameters, which is not compatible with the one in 'tslib'. Consider upgrading your version of 'tslib'.
```

For the most part it's impossible to tell that there's a problem - in fact, there is _no_ problem when I run this application in the test app that comes with `TSDX`. However, when I consume the library, this function appears to fail in interesting ways (basically, I just get a blank object instead of any actual data - so no "failure", just not what i'd expect).

I'm not particularly familiar with `tslib`, but `tsdx` has the oldest version dependency on `tslib` in my dependency graph. It's at 1.9.x and this function was installed at 1.10.x ([source](https://stackoverflow.com/a/59416828))

A few questions:
1. I tried simply changing my target ([source](https://stackoverflow.com/a/68188788)) - this didn't work. Should it have?
2. Is it possible to update the dependency in `tsdx` to `1.10`+? (it's now on a [2.x release](https://github.com/microsoft/tslib))

Any other guidance is very much appreciated!

**Comment by stephencweiss** [comment]
I also found this thread and am not sure if it's relevant: https://github.com/jaredpalmer/tsdx/issues/659

**Comment by rajeshias** [comment]
It says in the document to use __spreadArray instead
